tools/sigclient/pkg/query/functional.go

```diff
@@ -18,28 +18,64 @@
 package query
 
 import (
+	"bufio"
 	"os"
+	"path"
 	"path/filepath"
+
 	"strings"
 
 	log "github.com/sirupsen/logrus"
 )
 
 // Main function that tests all the queries
-func FunctionalTest(dest string, dataPath string) {
+func FunctionalTest(dest string, filePath string) {
+
+	// Check file exists
+	a, err := os.Stat(filePath)
+	if err != nil || a.IsDir() {
+		log.Fatalf("FunctionalTest: filePath: %v is not a file, err: %v", filePath, err)
+	}
+	_ = a
 
-	queryFiles, err := os.ReadDir(dataPath)
+	file, err := os.Open(filePath)
 	if err != nil {
-		log.Fatalf("FunctionalTest: Error reading directory: %v, err: %v", dataPath, err)
+		log.Fatalf("FunctionalTest: Error reading filePath: %v, err: %v", filePath, err)
 	}
+	defer file.Close()
+
+	baseDir, err := filepath.Abs(filepath.Dir(filePath))
+	if err != nil {
+		log.Fatalf("FunctionalTest: Error getting absolute path of directory of filePath: %v, err: %v", filePath, err)
+	}
+
+	scanner := bufio.NewScanner(file)
 
-	// validate JSON files
-	for _, file := range queryFiles {
-		if !strings.HasSuffix(file.Name(), ".json") {
-			log.Fatalf("FunctionalTest: Invalid file format: %v. Expected .json", file.Name())
+	qid := 1
+	// Read line by line
+	for scanner.Scan() {
+		queryFilePath := strings.TrimSpace(scanner.Text())
+		if queryFilePath == "" {
+			continue
 		}
+		// Validate JSON Files
+		if !strings.HasSuffix(queryFilePath, ".json") {
+			log.Fatalf("FunctionalTest: Invalid file format: %v. Expected .json", queryFilePath)
+		}
+
+		if !path.IsAbs(queryFilePath) {
+			queryFilePath = filepath.Join(baseDir, queryFilePath)
+		}
+
+		RunQuery(queryFilePath, qid, dest)
+		qid++
```

Comment:
after `Scan` call completes (aka it returns false and you exit from the for loop), you should call the err function exposed by the scanner to see if there was any error because of which it returned false.

Reply:
addressed in the latest commit.